Improve error management: fix error output stream and spelling

voodoovampire opened the pull request:
Fixes #9 (partial improvement)

**Changes:**
- Fixed spelling: "SUCCESFULLY" → "SUCCESSFULLY" in shader error message
- Changed error output from cout to cerr (proper error stream)
- Applied consistency across shader.cpp and model.cpp files

This addresses basic error management improvements as requested in issue #9.
OpenGL error checking and more advanced error handling can be added in follow-up PRs.

**KWoC Contribution**

dhanvithnayak committed:
add further development notice

voodoovampire committed:
Improve error management: fix spelling and use cerr for errors

- Fixed 'SUCCESFULLY' -> 'SUCCESSFULLY' in shader error message
- Changed error output from cout to cerr (proper error stream)
- Applied consistency across shader.cpp and model.cpp files

This addresses part of issue #9. More improvements (OpenGL error
checking, error utility class) can be added in follow-up PRs.

voodoovampire committed:
Implement distribution-ready builds (issue #10)

- Modified Makefile to copy shader resources to build directory
- Updated main.cpp to use relative paths (./shaders/) for shader files
- This allows the binary to find resources when run from build directory

The build/ directory can now be distributed with shaders included.

dhanvithnayak commented:
How is my commit in this pr lmao

voodoovampire committed:
Merge upstream/main and resolve conflicts

- Keep shader/model error fixes (cerr instead of cout)
- Adopt new CMake build system
- Update README.md with new documentation
- Merge new shader system with geometry/compute support

voodoovampire commented:
I've resolved the merge conflicts. The PR is ready for your review.

dhanvithnayak commented on src/main.cpp:
```diff
@@ -1,24 +1,20 @@
-// clang-format off
```
This is an essential marker, do not remove this line

dhanvithnayak commented:
You have most probably resolved merge conflicts with an outdated version of main. Please open a new, clean pull request from the latest version of main

voodoovampire committed:
Add Bullet Physics library as submodule

- Add bullet3 submodule to external/
- Update CMakeLists.txt to include Bullet
- Link BulletDynamics, BulletCollision, LinearMath libraries

voodoovampire committed:
Add Bullet Physics integration and test files

- Add bullet3 submodule
- Update main CMakeLists.txt for Bullet
- Add simple test program to verify installation
- Ready for physics implementation in Issue #12

voodoovampire commented:
Hellloooo there can you please check the commit

Zone-Infinity commented:
> Hellloooo there can you please check the commit

@voodoovampire As he said

> Please open a new, clean pull request from the latest version of main

This PR has 3 issues together lmao
